core/src/main/java/org/infinispan/distribution/ch/impl/HashFunctionPartitioner.java

```diff
@@ -35,6 +35,16 @@ public void init(HashConfiguration configuration) {
       init(configuration.numSegments());
    }
 
+   @Override
+   public void init(KeyPartitioner other) {
+      if (other instanceof HashFunctionPartitioner) {
+         HashFunctionPartitioner o = (HashFunctionPartitioner) other;
+         if (o.numSegments > 0) { // The other HFP has been initialized, so we can use it
```

Comment:
Not always... `init(HashConfiguration)` is called only when the cache is wired. The KeyPartitioner that's in the configuration before the cache is wired won't be initialized (e.g. if the CONFIG cache stores a configuration instance and the cache is wired from a copy).